src/listeners.js

```diff
@@ -0,0 +1,17 @@
+import { displayWeather } from './ui';
+
+function loadListeners() {
+  document.querySelector('.search-submit').addEventListener('click', () => {
+    const place = document.querySelector('.search').value;
+    const currentPlace = document
+      .querySelector('.place')
+      .textContent.split(',')[0];
+    if (place !== '' && place !== currentPlace) displayWeather(place);
+    document.querySelector('.search').value = '';
+  });
+  document.querySelector('.search').addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') document.querySelector('.search-submit').click();
```

Comment:
If you are up for experimenting, you could try to do keypress search with debounce mechanism. I.e. you write something and when you stop writing for 500 ms, api calls weather for entered text automatically.